python/fledge/services/core/api/plugins/remove.py

```diff
@@ -0,0 +1,165 @@
+from fledge.services.core import connect
+from fledge.common.storage_client.payload_builder import PayloadBuilder
+from fledge.common.common import _FLEDGE_PLUGIN_PATH, _FLEDGE_ROOT
+
+__author__ = "Rajesh Kumar"
+__copyright__ = "Copyright (c) 2019, Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_logger = logger.setup(__name__, level=logging.INFO)
+
+valid_plugin = ['north', 'south', 'filter', 'notificationDelivery', 'notificationRule']
+PYTHON_PLUGIN_PATH = _FLEDGE_ROOT+'/python/fledge/plugins/'
+C_PLUGINS_PATH = _FLEDGE_ROOT+'/plugins/'
+
+async def plugin_delete(request):
+    """
+    Remove plugin from fledge
+
+    '''
+    EndPoint: curl -X DELETE http://host-ip:port/fledge/plugins/{type}/{name}
+    '''
+    Example:
+        curl -X DELETE http://host-ip:port/fledge/plugins/south/sinusoid
+        curl -X DELETE http://host-ip:port/fledge/plugins/north/http_north
+        curl -X DELETE http://host-ip:port/fledge/plugins/filter/expression
+        curl -X DELETE http://host-ip:port/fledge/plugins/notificationDelivery/alexa
+        curl -X DELETE http://host-ip:port/fledge/plugins/notificationRule/Average
+    """
+    plugin_type = request.match_info.get('type', None)
+    name = request.match_info.get('name', None)
+    try:
+        plugin_type = str(plugin_type).lower() if not str(plugin_type).startswith('notification') else plugin_type
+        if plugin_type not in valid_plugin:
+            raise ValueError("Invalid plugin type.Please provide valid type:{}".format(valid_plugin))
+        installed_plugin = PluginDiscovery.get_plugins_installed(plugin_type, False)
+        _logger.info(f"installed plugin:{installed_plugin}")
+        if name not in [plugin['name'] for plugin in installed_plugin]:
+            raise KeyError("Invalid {} plugin name or plugin is not installed.".format(name))
+        if plugin_type in ['notificationDelivery', 'notificationRule']:
+            plugin_type = 'notify' if plugin_type == 'notificationDelivery' else 'rule'
+        get_tracked_plugins = await check_service_is_enabled_or_disabled(plugin_type, name)
+        _logger.info(get_tracked_plugins)
+        if get_tracked_plugins:
+            _logger.error("{} is being used.Purge is not possible in plugin enable state.".format(name))
+            raise RuntimeError("{} is being used.Purge is not possible in plugin enable state.".format(name))
+        res, log_path = purge_plugin(plugin_type, name)
+        if res != 0:
+            _logger.error("Something went wrong.Please check log:-{}".format(log_path))
+            raise RuntimeError("Something went wrong.Please check log:-{}".format(log_path))
+    except ValueError as ex:
+        return web.json_response({'error': '{}'.format(ex)}, status=400)
+    except KeyError as ex:
+        return web.json_response({'error': '{}'.format(ex)}, status=400)
+    except RuntimeError as ex:
+        return web.json_response({'error': '{}'.format(ex)}, status=400)
```

Comment:
raise web exception - i.e web.HTTPBadRequest with reason. Also I am not sure all cases with 400?